### ISSUE TITLE
Using DynamoDb DAX behind a proxy

### ISSUE THREAD

**Issue by gauravbrills** [aws/aws-sdk-java#1287]
Currently Client Config does not have ability to pass proxy params similar to ClientConfiguration for configuring Dax, This creates problem while working behind a corporate proxy .Example code 

```
public AmazonDynamoDB AmazonDynamoDB()
      {
         ClientConfig daxConfig = new ClientConfig().withRegion(Regions.US_EAST_1.getCurrentRegion())               
               .withEndpoints("test.cache.amazonaws.com:8111")//
               .withCredentialsProvider(new AWSStaticCredentialsProvider(this.amazonAWSCredentials()))//
              ;
         AmazonDaxClientBuilder daxClientBuilder = AmazonDaxClientBuilder.standard();
         daxClientBuilder.withClientConfiguration(daxConfig);
         AmazonDaxClient client = new ClusterDaxClient(daxConfig);
      
         AmazonDynamoDB dynamoDb = daxClientBuilder.build();
         return dynamoDb;
      }
```

Cant the api of ClientConfig be similar to ClientConfiguration where we can set proxy host and port . Or is there a way to do the same with ClientConfig.

Please Help!

**Comment by shorea** [comment]
I'll track down who owns this code and get back to you @gauravbrills

**Comment by gauravbrills** [comment]
@shorea  Thanks this is from the dax lib downloaded from http://dax-sdk.s3-website-us-west-2.amazonaws.com/ as mentioned in the docs .

**Comment by shorea** [comment]
@gauravbrills I've opened a ticket with the DAX team internally. I'm going to close this issue but we will continue to add correspondence as we get updates from the DAX team.

I did have a quick question. Are you expecting to be able to use an HTTP proxy or are you trying to use just a TCP proxy? My understanding is that DAX uses a custom, low level protocol directly over TCP so only a TCP proxy would be possible.
